pkg/pool-stable-phantom/contracts/StablePoolStorage.sol

```diff
@@ -0,0 +1,365 @@
+        _rateProvider5 = (rateProviders.length > 5) ? rateProviders[5] : IRateProvider(0);
+
+        _exemptFromYieldProtocolFeeToken0 = exemptFromYieldFlags[0];
+        _exemptFromYieldProtocolFeeToken1 = exemptFromYieldFlags[1];
+        _exemptFromYieldProtocolFeeToken2 = exemptFromYieldFlags[2];
+        _exemptFromYieldProtocolFeeToken3 = (rateProviders.length > 3) ? exemptFromYieldFlags[3] : false;
+        _exemptFromYieldProtocolFeeToken4 = (rateProviders.length > 4) ? exemptFromYieldFlags[4] : false;
+        _exemptFromYieldProtocolFeeToken5 = (rateProviders.length > 5) ? exemptFromYieldFlags[5] : false;
+    }
+
+    function _getTotalTokens() internal view virtual override returns (uint256) {
+        return _totalTokens;
+    }
+
+    function _getMaxTokens() internal pure override returns (uint256) {
+        // The BPT will be one of the Pool tokens, but it is unaffected by the Stable 5 token limit.
+        return StableMath._MAX_STABLE_TOKENS + 1;
+    }
+
+    function getBptIndex() public view returns (uint256) {
+        return _bptIndex;
+    }
+
+    // Convert from an index into an array including BPT (the Vault's registered token list), to an index
+    // into an array excluding BPT (usually from user input, such as amountsIn/Out).
+    // `index` must not be the BPT token index itself.
+    function _skipBptIndex(uint256 index) internal view returns (uint256) {
+        // Currently this is never called with an index passed in from user input, so this check
+        // should not be necessary. Included for completion (and future proofing).
+        _require(index != getBptIndex(), Errors.OUT_OF_BOUNDS);
+
+        return index < getBptIndex() ? index : index.sub(1);
+    }
+
+    // Convert from an index into an array excluding BPT (usually from user input, such as amountsIn/Out),
+    // to an index into an array excluding BPT (the Vault's registered token list).
+    // `index` must not be the BPT token index itself, if it is the last element, and the result must be
+    // in the range of registered tokens.
+    function _addBptIndex(uint256 index) internal view returns (uint256 indexWithBpt) {
+        // This can be called from an index passed in from user input.
+        indexWithBpt = index < getBptIndex() ? index : index.add(1);
+
+        _require(indexWithBpt < _totalTokens && indexWithBpt != getBptIndex(), Errors.OUT_OF_BOUNDS);
+    }
+
+    /**
+     * @dev Remove the item at `_bptIndex` from an arbitrary array (e.g., amountsIn).
+     */
+    function _dropBptItem(uint256[] memory amounts) internal view returns (uint256[] memory) {
+        uint256[] memory amountsWithoutBpt = new uint256[](amounts.length - 1);
+        for (uint256 i = 0; i < amountsWithoutBpt.length; i++) {
+            amountsWithoutBpt[i] = amounts[i < getBptIndex() ? i : i + 1];
+        }
+
+        return amountsWithoutBpt;
+    }
+
+    // Helpers
```

Comment:
This line has been stranded a bit due to reordering.

Reply:
Yes - I moved it up to the virtual supply section. It's similar to a helper function in StablePoolStorage, and I considered trying to put it there: but then I would have to move other virtual supply stuff, and as it has nothing to do with immutable storage, I think that should remain in the main contract.

This made me realize StablePoolStorage is a bit of a mess, so I've reordered that as well.